enterprise/server/remote_execution/copy_on_write/copy_on_write.go

```diff
@@ -921,82 +953,74 @@ func (m *Mmap) WriteAt(p []byte, off int64) (n int, err error) {
 }
 
 func (m *Mmap) Sync() error {
-	if !m.safeReadMapped() {
-		return nil
-	}
 	m.mu.RLock()
 	defer m.mu.RUnlock()
+	if m.data == nil {
+		return nil
+	}
 	return unix.Msync(m.data, unix.MS_SYNC)
 }
 
-func (m *Mmap) Close() error {
+// Unmap unmaps the chunk without marking it closed.
+// It returns nil if the chunk is already unmapped.
+func (m *Mmap) Unmap() error {
 	m.mu.Lock()
 	defer m.mu.Unlock()
-	m.closed = true
-	if m.source == snaputil.ChunkSourceUnmapped {
+	if m.data == nil {
 		return nil
 	}
-	m.source = snaputil.ChunkSourceUnmapped
 	n := len(m.data)
 	if err := syscall.Munmap(m.data); err != nil {
 		return err
 	}
+	m.data = nil
 	metrics.COWSnapshotMemoryMappedBytes.Set(float64(
 		atomic.AddInt64(&mmappedBytes, int64(-n)),
 	))
+	// TODO: this causes issues because when we unmap then remap due to LRU
+	// eviction, we lose the remote/filecache source information, since Fetch()
+	// won't be called again. Is it safe to just delete this line?
+	// m.source = snaputil.ChunkSourceUnmapped
```

Comment:
Do you want this in the PR?

Reply:
woops - removed this chunk, including the line setting m.source = Unmapped